packages/sync/src/utils/calculatePageLimit.ts

```diff
@@ -0,0 +1,49 @@
+interface DynamicLimiter {
+  initialLimit: number;
+  minLimit: number;
+  maxLimit: number;
+  optimalTimePerPageMs: number;
+  maxLimitChangePerPage: number;
+}
+
+// TODO: Move to config model RN-1668
+export const DYNAMIC_LIMITER = {
+  initialLimit: 10, // start relatively low then grow upward
+  minLimit: 1,
+  maxLimit: 10000,
+  optimalTimePerPageMs: 2000, // aim for 2 seconds per page
+  maxLimitChangePerPage: 0.2, // max 20% increase/decrease from page to page
+};
+
+// Set the current page limit based on how long the previous page took to complete.
+export const calculatePageLimit = (
```

Comment:
Do you think the dynamic limiter is worth it now we are only using it for push? Also wonder if you considered using streaming for push?

Reply:
I think the answer to both is probably "yes" btw!